### PR TITLE
Uninstall agent even if the objects are in an inconsistent state

### PR DESCRIPTION
**Pull request by josecv**
Signed-off-by: Jose Cortes <josecortes@datawire.io>

## Description

A few sentences describing the overall goals of the pull request's commits.

## Checklist

 - [x] I made sure to update `./CHANGELOG.md`.
 - [ ] I made sure to either submit a docs PR, or tell Matt about the necessary documentation changes.
 - [ ] My change is adequately tested.
 - [ ] I updated `DEVELOPING.md` with any any special dev tricks I had to use to work on this code efficiently.

### CODE REVIEW

File: pkg/install/util.go [hunk 1]
```diff
@@ -40,3 +42,39 @@ func ObjErrorf(obj kates.Object, format string, args ...interface{}) error {
...
+
+func (e *AlreadyUndone) Unwrap() error {
+	return e.err
+}
+
+func NewAlreadyUndone(err error, msg string) error {
+	return &AlreadyUndone{err, msg}
+}
+
+// IsAlreadyUndone returns whether the given error -- possibly a multierror -- indicates that all actions have been undone.
```
Comment (LukeShu) [inline pkg/install/util.go:64]: My high school journalism teacher would have yelled at you for putting spaces round the `--` em-dashes :P

File: pkg/install/util.go [hunk 2]
```diff
@@ -40,3 +42,39 @@ func ObjErrorf(obj kates.Object, format string, args ...interface{}) error {
...
+	if !errors.As(err, &multi) {
+		return false
+	}
+	for _, err := range multi.WrappedErrors() {
+		if !errors.As(err, &undone) {
+			return false
+		}
+	}
+	return true
+}
```
Comment (LukeShu) [inline pkg/install/util.go:80]: Looking at `go-multierror`, I'm like 90% sure that this is unnecessary and that you can just use `errors.As(err, &undone)`

**Review by LukeShu — APPROVED** [review]
A nitpick with `IsAlreadyUndone` that you can choose to ignore or tell me I'm wrong.

Resolve the merge conflicts, and it looks good!

File: pkg/install/util.go (hunk 2 above)
Comment (josecv) [reply pkg/install/util.go:80]: I don't think so -- this is slightly different. My understanding of `go-multierror` is that `errors.As` will return "is _any_ error in this multierror an AlreadyUndone", whereas here we want _all_ errors in the multierror to eventually resolve back to an `AlreadyUndone`

**Pull request merged**